### ISSUE TITLE
Properly wait for namespace creation to stabilize in tests

### ISSUE THREAD

**Issue by thomastaylor312** [deislabs/krustlet#403]
Currently in our integration tests, we are just doing [a delay](https://github.com/deislabs/krustlet/pull/396#discussion_r495082143) to give the controller enough time to create the token for the `default` service account. However, we should do this more smartly. Ideally, we should do a watch (on exactly what will be part of this issue, probably something like the status of the `ServiceAccount` or the tokens in the namespace) to wait for the token to be created

**Comment by nitishm** [comment]
@thomastaylor312 I would love to help out with this albeit with some hand holding since I have just started with Rust over the past few weeks. Please let me know if this is something that might be suited for a beginner to take on (I have experience writing controllers for Kubernetes in Go, so am aware of how the kube client is supposed to work)

**Comment by thomastaylor312** [comment]
Yeah, if you have that deep of background in Kubernetes, this task should be just fine for a beginner. Given that k8s 1.21+ uses the `projected` volume to inject tokens, this is now less of a problem, but will still effect 1.20. 

Please let me know if you have any issues, as running the e2e tests can be difficult at times
